main.cpp

```diff
@@ -39,19 +40,57 @@ struct Point
         str += std::to_string(y);
         str += " }";
         return str;
-    }
+    }
+
 private:
     float x{0}, y{0};
 };
 
 template<typename Type>
 struct Wrapper
 {
-    Wrapper(Type&& t) : val(std::move(t)) 
+    Wrapper(Type&& t) : val(std::move(t) ) 
     { 
         std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
-    }
-};
+    }
+
+    //print() function added to Wrapper stub
+    void print() const;
+
+private:
+    Type val;
+};
+
+//wrapper print() function definition
+template<typename Type>
+void Wrapper<Type> :: print() const
+{
+    std::cout << "Wrapper::print(" << val << ")" << std::endl;
+}
+
+//wrapper class point speciailization
+template<>
+void Wrapper<Point>::print() const
+{
+    std::cout << val.toString() << std::endl;
+}
+
+//variadic-template version of variadicHelper
+template<typename Type, typename ...Args>
+void variadicHelper(Type first, Args ...everything)
+{
+    Wrapper<Type> wrap(std::forward<Type>(first));
+    wrap.print();
+    variadicHelper(std::forward<Args>(everything)...);
+}
+
+//single-template-parameter version of variadicHelper
+template<typename Type>
+void variadicHelper(Type first)
+{
+    Wrapper<Type> wrap(std::forward<Type>(first));
+    wrap.print();
+}

```

Comment:
Project9.1 challenge:
There is another recursive solution.  it is possible for `Args ...` to be empty.
your task: remove your implementation for instruction `4)` and implement this alternate solution.
implement it **after** your existing `variadicHelper` function.
hint: what does the **recursive call** actually look like when called with an empty Variadic Parameter list?

- [ ] revise